src/Compilers/CSharp/Portable/Symbols/MetadataOrSourceAssemblySymbol.cs

```diff
@@ -21,12 +21,44 @@ internal abstract class MetadataOrSourceAssemblySymbol
         /// </summary>
         /// <remarks></remarks>
         private NamedTypeSymbol[] _lazySpecialTypes;
+        private ThreeState _lazyRuntimeSupportsDefaultInterfaceImplementation = ThreeState.Unknown;
 
         /// <summary>
         /// How many Cor types have we cached so far.
         /// </summary>
         private int _cachedSpecialTypes;
 
+        internal override bool RuntimeSupportsDefaultInterfaceImplementation
+        {
+            get
+            {
+                if ((object)CorLibrary == this)
+                {
+                    if (!_lazyRuntimeSupportsDefaultInterfaceImplementation.HasValue())
+                    {
+                        // PROTOTYPE(DefaultInterfaceImplementation): Implement real detection if the feature is supported by the runtime
+                        // For now we assume that it is supported by default
+                        _lazyRuntimeSupportsDefaultInterfaceImplementation = ThreeState.True;
```

Comment:
> Don't we want this write to be immediately visible to other threads?

In case of a very small chance of that not happening, the behavior is still going to be correct. This is a common pattern in both compilers around structures that are guaranteed to be written atomically.